Application/EdFi.Ods.AdminApi/Features/ClaimSets/ResourceClaims/EditAuthStrategy.cs

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using AutoMapper;
+using EdFi.Ods.AdminApi.Infrastructure;
+using EdFi.Ods.AdminApi.Infrastructure.ClaimSetEditor;
+
+namespace EdFi.Ods.AdminApi.Features.ClaimSets.ResourceClaims;
+
+public class EditAuthStrategy : IFeature
+{
+    public void MapEndpoints(IEndpointRouteBuilder endpoints)
+    {
+        AdminApiEndpointBuilder.MapPost(endpoints, "/claimsets/{claimsetid}/resourceclaims/{resourceclaimid}/overrideauthstrategy", HandleOverrideAuthStrategies)
+       .WithDefaultDescription()
+       .BuildForVersions(AdminApiVersions.V2);
+
+        AdminApiEndpointBuilder.MapPost(endpoints, "/claimsets/{claimsetid}/resourceclaims/{resourceclaimid}/resetauthstrategies", HandleResetAuthStrategies)
+        .WithDefaultDescription()
+        .BuildForVersions(AdminApiVersions.V2);
+    }
+
+    public async Task<IResult> HandleOverrideAuthStrategies(OverrideAuthStategyOnClaimSetValidator validator,
+      OverrideDefaultAuthorizationStrategyCommand overrideDefaultAuthorizationStrategyCommand, IMapper mapper,
+      OverrideAuthStategyOnClaimSetRequest request, int claimsetid, int resourceclaimid)
+    {
+        request.ClaimSetId = claimsetid;
+        request.ResourceClaimId = resourceclaimid;
+        await validator.GuardAsync(request);
+        var model = mapper.Map<OverrideAuthStategyOnClaimSetModel>(request);
+        overrideDefaultAuthorizationStrategyCommand.ExecuteOnSpecificAction(model);
+
+        return Results.Ok();
+    }
+
+    public async Task<IResult> HandleResetAuthStrategies(IGetResourcesByClaimSetIdQuery getResourcesByClaimSetIdQuery,
+        IAuthStrategyResolver strategyResolver,
+        UpdateResourcesOnClaimSetCommand updateResourcesOnClaimSetCommand,
+        IMapper mapper, int claimsetid, int resourceclaimid)
+    {
+        var resourceClaims = getResourcesByClaimSetIdQuery.AllResources(claimsetid);
+        if (!resourceClaims.Any(rc=>rc.Id == resourceclaimid))
+        {
+            throw new NotFoundException<int>("ResourceClaim", resourceclaimid);
+        }
+        else
+        {
+            foreach (var resourceClaim in resourceClaims)
+            {
+                if (resourceClaim.Id == resourceclaimid)
+                {
+                    resourceClaim.DefaultAuthStrategiesForCRUD = Array.Empty<AuthorizationStrategy>();
+                    resourceClaim.AuthStrategyOverridesForCRUD = Array.Empty<AuthorizationStrategy>();

```

Comment:
@jagudelo-gap 
We should not clear DefaultAuthStrategiesForCRUD. We should just clear AuthStrategyOverridesForCRUD

Reply:
@CSR2017 It won't be clear at all, It will set the default values again. Line 59 will get the default values to populate again the property DefaultAuthStrategiesForCRUD

Reply:
@CSR2017 
I did a refactor for this code